middleware.js

```diff
@@ -1,4 +1,45 @@
 var http = require('http');
+var url = require('url');
 
-exports.createServer = function () {
+module.exports = {
+  createServer: function () {
+    var routes = [];
+
+    return {
+      use: function (path, handler) {
+        if (typeof path === 'function') {
+          handler = path;
+          path = '*';
+        }
+        routes.push({ [path]: handler });
```

Comment:
If you save this `path-handler` relation like `{path: path, handler: handler}`, you can write more elegant code in [L27](https://github.com/weflex/backend-challenge/pull/4/files#diff-27e39f85634d9276821c67a9bbf764b4R27).